src/coreclr/vm/jitinterface.cpp

```diff
@@ -7210,7 +7232,8 @@ bool getILIntrinsicImplementationForRuntimeHelpers(MethodDesc * ftn,
             || methodTable == CoreLibBinder::GetClass(CLASS__INTPTR)
             || methodTable == CoreLibBinder::GetClass(CLASS__UINTPTR)
             || methodTable == CoreLibBinder::GetClass(CLASS__RUNE)
-            || methodTable->IsEnum())
+            || methodTable->IsEnum()
+            || IsBitwiseEquatable(typeHandle, methodTable))
```

Comment:
Could you please make a matching change for the AOT compilers?

https://github.com/dotnet/runtime/blob/ab2d195f169a08bf8eb978b472b2d3aca61b0a46/src/coreclr/tools/Common/TypeSystem/IL/Stubs/RuntimeHelpersIntrinsics.cs#L54

Reply:
We have the algorithm in https://github.com/dotnet/runtime/blob/fa42038b64236a49b0a74776ec9a8b321b95b98c/src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/CompilerTypeSystemContext.ValueTypeMethods.cs#L116 but it's currently not callable from RuntimeHelpersIntrinsics.cs.

I would pull that out into a separate static method in a separate file (without all the caching) and change that place and this place to call the static method.

We probably don't need this to be behind a cache for IsBitwiseEquatable.

Reply:
Thanks. Will do.

Reply:
I made some changes locally, but haven't been able to successfully build (sent you a note in Teams).  In the meantime I pushed up the changes just to see how CI feels about it and to get your eyes on it in case I misinterpreted your feedback.